flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -817,7 +817,12 @@ protected void runWithArgs(String[] args, String terminateAfterString, String[]
 				}
 			}
 			// check output for the expected terminateAfterString.
-			testPassedFromLog4j = logMessageSupplier.get().stream().anyMatch(s -> s.contains(terminateAfterString));
+			testPassedFromLog4j = logMessageSupplier.get().stream()
+				.filter(s -> s.contains(terminateAfterString))
+				.limit(1)
```

Comment:
will it not throw if the stream is empty after the filtering?

Reply:
maybe just good old loop :)

Reply:
I see no such behavior documented in the javadocs, it just limits the max size. Minimum size should be irrelevant.